xbmc/interfaces/builtins/ApplicationBuiltins.cpp

```diff
@@ -0,0 +1,159 @@
+{
+  // Detects if file is zip or rar then extracts
+  std::string strDestDirect;
+  if (params.size() < 2)
+    strDestDirect = URIUtils::GetDirectory(params[0]);
+  else
+    strDestDirect = params[1];
+
+  URIUtils::AddSlashAtEnd(strDestDirect);
+
+  CFileItemList items;
+  CFileItemPtr ptr(new CFileItem());
+  CURL archpath = URIUtils::CreateArchivePath(URIUtils::GetExtension(params[0]).substr(1), CURL(params[0]), "");
+  ptr->SetURL(archpath);
+  ptr->Select(true);
+  CFileOperationJob job(CFileOperationJob::ActionCopy, items, strDestDirect);
+  if (!job.DoWork())
+    CLog::Log(LOGERROR, "XBMC.Extract, Error extracting archive");
+
+  return 0;
+}
+
+/*! \brief Mute volume.
+ *  \param params (ignored)
+ */
+static int Mute(const std::vector<std::string>& params)
+{
+  g_application.ToggleMute();
+
+  return 0;
+}
+
+/*! \brief Notify all listeners on announcement bus.
+ *  \param params The parameters.
+ *  \details params[0] = sender.
+ *           params[1] = data.
+ *           params[2] = JSON with extra parameters (optional).
+ */
+static int NotifyAll(const std::vector<std::string>& params)
+{
+  CVariant data;
+  if (params.size() > 2)
+    data = CJSONVariantParser::Parse((const unsigned char *)params[2].c_str(), params[2].size());
+
+  ANNOUNCEMENT::CAnnouncementManager::Get().Announce(ANNOUNCEMENT::Other, params[0].c_str(), params[1].c_str(), data);
+
+  return 0;
+}
+
+/*! \brief Set volume.
+ *  \param params the parameters.
+ *  \details params[0] = Volume level.
+ *           params[1] = "showVolumeBar" to show volume bar (optional).
+ */
+static int SetVolume(const std::vector<std::string>& params)
+{
+  float oldVolume = g_application.GetVolume();
+  float volume = (float)strtod(params[0].c_str(), NULL);
+
+  g_application.SetVolume(volume);
+  if(oldVolume != volume)
+  {
+    if(params.size() > 1 && StringUtils::EqualsNoCase(params[1], "showVolumeBar"))
+    {
+      CApplicationMessenger::Get().ShowVolumeBar(oldVolume < volume);
+    }
+  }
+
+  return 0;
+}
+
+/*! \brief Toggle debug info.
+ *  \param params (ignored)
+ */
+static int ToggleDebug(const std::vector<std::string>& params)
+{
+  bool debug = CSettings::Get().GetBool("debug.showloginfo");
+  CSettings::Get().SetBool("debug.showloginfo", !debug);
+  g_advancedSettings.SetDebugMode(!debug);
+
+  return 0;
+}
+
+/*! \brief Toggle DPMS state.
+ *  \param params (ignored)
+ */
+static int ToggleDPMS(const std::vector<std::string>& params)
+{
+  g_application.ToggleDPMS(true);
+
+  return 0;
+}
+
+/*! \brief Send a WOL packet to a given host.
+ *  \param params The parameters.
+ *  \details params[0] = The MAC of the host to wake.
+ */
+static int WakeOnLAN(const std::vector<std::string>& params)
+{
+  g_application.getNetwork().WakeOnLan((char*)params[0].c_str());
+
+  return 0;
+}
+
+CBuiltins::CommandMap CApplicationBuiltins::GetOperations() const
+{
+  return {
+           {"extract", {"Extracts the specified archive", 1, Extract}},
```

Comment:
minor nitpick: mind using the same nice idention of the second value like you did in all the other files?